go-client/examples/basic_query/main.go

```diff
@@ -0,0 +1,100 @@
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/deephaven/deephaven-core/go-client/client"
+	"github.com/deephaven/deephaven-core/go-client/examples/common"
+)
+
+// This example shows how to use the powerful query system,
+// which allows you to perform an arbitrary number of table operations in a single request.
+// It can even create new tables as part of a query,
+// and the interface is more convenient than doing each of the table operations serially.
+func main() {
+	// A context is used to set timeouts and deadlines for requests or cancel requests.
+	// If you don't have any specific requirements, context.Background() is a good default.
+	ctx := context.Background()
+
+	// When starting a client connection, note that the client script language "python"
+	// must match the language the server was started with,
+	// even if we aren't using any scripts.
+	cl, err := client.NewClient(ctx, "localhost", "10000", "python")
+	if err != nil {
+		fmt.Println("error when connecting to localhost port 10000:", err.Error())
+		return
+	}
+	defer cl.Close()
+
+	// First, let's create some example data to manipulate.
+	sampleRecord := common.GetExampleRecord()
+	// Note that Arrow records must eventually be released.
+	defer sampleRecord.Release()
+
+	fmt.Println("Data Before:")
+	fmt.Println(sampleRecord)
+
+	// Now we upload the record so that we can manipulate its data using the server.
+	// We get back a TableHandle, which is a reference to a table on the server.
+	baseTable, err := cl.ImportTable(ctx, sampleRecord)
+	if err != nil {
+		fmt.Println("error when uploading table:", err.Error())
+		return
+	}
+	// Table handles should be released when they are no longer needed.
+	defer baseTable.Release(ctx)
+
+	// Now, let's start building a query.
+	// Maybe I don't like companies whose names are too long or too short, so let's filter in only the ones in the middle.
+	// Note that we use baseTable.Query() to start building a new query from an existing table handle.
+	// The Query() method returns a QueryNode.
+	midStocks := baseTable.Query().
+		Where("Ticker.length() == 3 || Ticker.length() == 3")
+
+	// We can use the query system to create completely new tables too.
+	// Let's make a table whose columns are powers of ten.
+	// Again, EmptyTableQuery() returns a QueryNode.
+	powTenTable := cl.
+		EmptyTableQuery(10).
+		Update("Magnitude = (int)pow(10, ii)")
+
+	// What if I want to bin the companies according to the magnitude of the Vol column?
+	// Query methods can take other query nodes as arguments to build up arbitrarily complicated requests,
+	// so we can perform an as-of join between two query nodes just fine.
+	magStocks := midStocks.
+		AsOfJoin(powTenTable, []string{"Vol = Magnitude"}, nil, client.MatchRuleLessThanEqual)
+
+	// And now, we can execute the entire query we have built using only a single request.
+	// Note that we can retrieve multiple tables from anywhere in the query.
+	tables, err := cl.ExecQuery(ctx, midStocks, magStocks)
```

Comment:
I was confused when I first looked at this.  Possibly `ExecQueries` is a better name, since there can be multiple inputs and outputs.

Reply:
I ended up renaming it to `ExecBatch`, to separate the query (the actual operations) from the execution method (batch or serial).